tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -33,10 +33,23 @@ def parse_datafile(datafile):
         return errors
 
     field_values = schema_defs.header.get_field_values_by_names(header_line,
-                                                                {"encryption", "tribe_code", "state_fips"})
+                                                                {"encryption", "tribe_code", "state_fips", "update"})
 
-    # Validate tribe code in submission across program type and fips code
     generate_error = util.make_generate_parser_error(datafile, 1)
+
+    # Validate header update code is "D"
+
+    update_indicator_is_valid, update_indicator_error = validators.validate_update_indicator(field_values["update"],
```

Comment:
this is mostly just a point of discussion rather than a request for changes

with the inclusion of this file pre-check, we have four nearly-identical operations that all throw a cat1 error, kill parsing, and reject the file. as i understand it, this idea was initially a one-off but we've gone on to use the same pattern many times. i think a list of these functions that could be looped through would be more concise and allow us to handle the pre-check case more consistently

Reply:
I like the sound of that also. Would like to see what that would look like implemented.